File: singly_linked_list/recipe-01/cxx/test_detele_tail.cpp
#include "singly_linked_list.hpp"
#include <iostream>
#include <string>

void test_delete_tail() {
    std::cout << "----------------------------" << __func__ << "---------------------------\n";

    SinglyLinkedList<std::string> words;
    words.Append("foo");
    words.Append("bar");
    words.Append("bim");
    words.Append("baz");
    words.Append("quux");

    std::cout << "The origin list" << '\n';
    std::cout << "size: " << words.Count() << '\n';
    for (const auto& word : words)
        std::cout << "data: " << word << '\n';

    std::cout << "Delete the last item in the list" << '\n';
    words.Delete("quux");
    std::cout << "size: " << words.Count() << '\n';
    for (const auto& word : words)
        std::cout << "data: " << word << '\n';

    std::cout << "append another item" << '\n';
    words.Append("Quux");
    std::cout << "size: " << words.Count() << '\n';
    for (const auto& word : words)
        std::cout << "data: " << word << '\n';
}

int main()
{
    test_delete_tail();
    return 0;
}

File: singly_linked_list/recipe-01/python/test_detele_tail.py
from singly_linked_list import SinglyLinkedList

def test_delete_tail():
    print("----------------------------test_delete_tail---------------------------");

    numbers = SinglyLinkedList()
    for i in range(5):
        numbers.Append(i)
    
    print("The origin list");
    for number in numbers:
        print("data: {}".format(number))

    print("Delete the last item in the list")
    numbers.Delete(4)
    print("size: {}".format(numbers.count))
    for number in numbers:
        print("data: {}".format(number))

    print("append another item")
    numbers.Append(6)
    print("size: {}".format(numbers.count))
    for number in numbers:
        print("data: {}".format(number))

def main():
    test_delete_tail()

main()
